zulipterminal/ui_tools/views.py

```diff
@@ -1178,7 +1189,23 @@ def _fetch_user_data(controller: Any, user_id: int) -> Dict[str, Any]:
             if data["last_active"]:
                 display_data["Last active"] = data["last_active"]
 
-        return display_data
+        custom_profile_data = data["custom_profile_data"]
+
+        if custom_profile_data:
+            for field in custom_profile_data:
+                # Get label, type, order and value of field data
+                if field["type"] == 6:  # Person picker
+                    user_names = []
+                    for user in field["value"]:
+                        user_data = controller.model._all_users_by_id.get(user)
```

Comment:
This should use `user_name_from_id` to avoid accessing the implementation.

I'd be open to adjusting the error behavior in that method to make this cleaner; perhaps providing an optional value different would return that instead of raising the exception?
(it looks like there are some other calling sites we could improve in any case)

I suspect that would enable making this into a simple list comprehension, or even generator expression with the join.